### ISSUE TITLE
Only show checklists which are children to the currently selected folder

### ISSUE THREAD

**Issue by Troberg** [delashum/obsidian-checklist-plugin#74]
This would provide an effective, yet simple way to filter checklists intelligently, even if you have many unrelated checklists in your structure.

For example:

* To see all checklists, select top level.
* To see all development checklists, select the development folder which you have all your projects in.
* To see checklists from a single projects, select the base level for that project.

**Comment by delashum** [comment]
You can use the `Include Files` option in the plugin settings to filter which files are included in the search. Does this accomplish what you are looking for?

**Comment by Troberg** [comment]
Well, yes and no. It would work, but having to go into settings all the time would be cumbersome.

**Comment by mochsner** [comment]
_Disclaimer: I'm not a maintainer - just another curious guy that enjoys this plugin_

To better understand what you're going for @Troberg, would you be fine with an enabled/disabled setting that would toggle this new more narrow scope for the checklist view? So, only when you have the setting enabled, the checklist view will always based on the scope of where you're selecting in the folder (or file) left side-bar.

I imagine something like this at its core could be useful functionality to add, with the possibility down the road for usability to be improved (such as right clicking to change the checklist scope to a folder without this "always on" setting needing to be enabled)

**Comment by Troberg** [comment]
Yep, that would work.

The thing for me is that I keep all my projects in the same obsidian folder. This means that I have subdirectorites for programming project, home projects, motorcycle projects, workshop projects and so on, all with their own todos. If I view them all, it's thousands of items (I have a lot of todos which are more like "could do"...), and usually, I'm just looking for stuff to do in one category.

**Comment by delashum** [comment]
What does it mean to select a folder or subfolder? Do you mean only parse sibling files of the file you currently have focused?

Another possibility would be to add command actions to quickly change settings. What do you think about that as a solution?

**Comment by Troberg** [comment]
Parse siblings all the way down. Looking at only one file isn't useful, that's pretty much what is done without this plugin.

Command actions would be a compromise. Better than today, more awkward than automatic selection.

**Comment by mochsner** [comment]
Yeah buttons are a pain, and since this is all experimental I'd argue that a command is best.

Side-note, I had a similar thought/want for my workflow the other week (similar issue as @delashum  with wanting to focus on specific "groupings" of tags). I would've loved the ability to just do a string filter of tags. Of course, this would be great for the todos themselves as well, but if it were possible to do the each separately.

I'm curious, @Troberg - would filtering based on the way you use your tags be useful at all to you based on your workflow? For me, I use `#todo/_work`, `#todo/_home`, `#todo/_devmeeting`, etc. So that's why it'd be great. Also if you don't use or weren't aware of using tags (or rather, subtags) like this, I highly recommend it.

**Comment by Troberg** [comment]
For me, it wouldn't help much with tags. That would mean that I would need to basically duplicate the directory structure in tags.

**Comment by mochsner** [comment]
Cool yeah that's what I figured - thanks for confirming!
